### ISSUE TITLE
XML parsing does not quit with an error for invalid XML source

### ISSUE THREAD

**Issue by Kopfbremse** [mikefarah/yq#1448]
**Version of yq**: 4.30.4
**Operating system**: linux
**Installed via**: binary release

**Bug description**
XML parsing does not quit for invalid XML source.
This can produce strange conversions results

**Input**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML: text_outside_1<Tag1 />text_outside_2<Tag2>Kling Klong</Tag2>text_outside_3
  time: 09:19:00
```

**Command**
```bash
yq --input-format xml
```

**Output**
```yaml
+content: |-
  text_outside_3
    time: 09:19:00
Tag1:
Tag2: Kling Klong
```

**Expected error**
```yaml
XML is not valid
```

**Comment by mikefarah** [comment]
Weird - I would have though the parser I'm using would throw an error. I can put my own check in though 👍🏼  - hopefully won't break anything else. Basically checking that there's no raw content outside a root node. Will fix in next release

**Comment by mikefarah** [comment]
Fixed in 4.30.5

**Comment by Kopfbremse** [comment]
thanks for the quick fix
